### DISCUSSION TITLE
and operator  in where clause

### DISCUSSION THREAD

**Discussion by AnouarMc** [ujjwalguptaofficial/JsStore#288] [Q&A] [answered]
Hi I have a table that looks like this: 

![image](https://user-images.githubusercontent.com/40692106/191143006-0f89f5cc-588d-42d6-9138-1e99a0804f8f.png)

I want to execute this query:

`SELECT * FROM dates WHERE due != "" AND due <= "2022-09-20"`

So that I can get "2022-09-20" and "2022-09-19"

when I execute this  code: 

`await window.JS.select({
  from: "words",
  where: {
    due: { "<=": "2022-09-20" },
    due: { "!=": "" }
  }
});`

I get "2022-09-21" , "2022-09-20" and "2022-09-19" 

If I changed the order inside the WHERE like this:

`await window.JS.select({
from: "words",
  where: {
    due: { "!=": "" },
    due: { "<=": "2022-09-20" }
  }
});`

I get "2022-09-20" , "2022-09-19" and "" (empty string)

How can I achieve the above query?

Thanks

**Comment by ujjwalguptaofficial** [comment, marked as answer]
your date value is stored as string, so you won't be able to compare it with another date. 

or do you want to string comparision only ?

if yes - then use array query for querying multiple value with same property. take a look at some query example - https://jsstore.net/tutorial/adv-sql-example/

**Reply by AnouarMc** [reply]
Thank you for your quick response.

Yes, I want only string comparison I store dates in "yyyy-MM-dd" format so even string comparison gives the intended results in SQL.

How can I achieve this: 

`SELECT * FROM dates WHERE due != "" AND due <= "2022-09-20"`

I could not find in the provided link an example of a query on same the column

**Reply by ujjwalguptaofficial** [reply]
@AnouarMc here is equivalent query - 

```
connection.select({
    from: "Table_Name",
    where: [
        {
            due:  { "!=": "" }
        },
        {
            due: { "<=": "2022-09-20" }
        }
    ]
});

```

the trick is to use array to specify multiple query.

**Reply by AnouarMc** [reply]
Thanks a lot. that what I was looking for. :+1:
